File: Lista de exercicio/projetoJAVA/src/projetojava/questao3.java
package projetojava;

class ContaBancaria2 {
    private double saldo;
    String titular;

    ContaBancaria2(String titular, double saldoInicial) {
        this.titular = titular;
        this.saldo = saldoInicial;
    }

    void depositar(double quantia) {
        saldo += quantia;
    }

    void sacar(double quantia) {
        if (quantia <= saldo) {
            saldo -= quantia;
        } else {
            System.out.println("Saldo insuficiente");
        }
    }

    String exibirSaldo() {
        return "Saldo atual: R$" + saldo;
    }
}

public class questao3 {
    public static void main(String[] args) {
        ContaBancaria2 conta = new ContaBancaria2("Maria", 500);
        conta.depositar(200);
        System.out.println(conta.exibirSaldo());
        conta.sacar(100);
        System.out.println(conta.exibirSaldo());
        conta.sacar(700);
    }
}

File: Lista de exercicio/projetoPYTHON/questao3/questao3.py
class ContaBancaria:
    def __init__(self, titular, saldo_inicial):
        self.__saldo = saldo_inicial
        self.titular = titular

    def depositar(self, quantia):
        self.__saldo += quantia

    def sacar(self, quantia):
        if quantia <= self.__saldo:
            self.__saldo -= quantia
        else:
            print("Saldo insuficiente")

    def exibir_saldo(self):
        return f"Saldo atual: R${self.__saldo}"

#Exemplo de uso:

def executar_questao3():
    conta = ContaBancaria("Maria", 500)
    conta.depositar(200)
    print(conta.exibir_saldo())
    conta.sacar(100)
    print(conta.exibir_saldo())
    conta.sacar(700) 